src/modules/voxin.c

```diff
@@ -0,0 +1,1955 @@
+			g_strdup(from->name);
+		for (i=0; to->name[i]; i++) {
+			to->name[i] = tolower(to->name[i]);
+		}
+	}
+	{ /* set language: language identifier (lower case) + variant/dialect (all caps) */
+		if (*from->variant) {
+			size_t len = strlen(from->lang);
+			int i;
+			to->language = g_strdup_printf("%s-%s", from->lang, from->variant);
+			for (i=len; to->language[i]; i++) {
+				to->language[i] = toupper(to->language[i]);
+			}
+		} else {
+			to->language = g_strdup(from->lang);
+		}
+	}
+	to->variant = g_strdup("none");
+	
+	{ /* log the 'from' argument */
+		size_t size = 0;
+		if (!voxToString(from, NULL, &size)) {
+			gchar *str = g_malloc0(size);
+			if (!voxToString(from, str, &size)) {
+				DBG(DBG_MODNAME "from: %s", str);
+			}
+			g_free(str);
+		}
+	}
+	DBG(DBG_MODNAME "to: name=%s, variant=%s, language=%s", to->name, to->variant, to->language);
+	return TRUE;
+}
+
+static gboolean alloc_voice_list()
+{
+	DBG(DBG_MODNAME "ENTER %s", __func__);
+	int i = 0;	
+
+	/* obtain the list of installed voices */
+	number_of_voices = 0;
+	if (voxGetVoices(NULL, &number_of_voices) || !number_of_voices) {
+		return FALSE;
+	}
+
+	voices = g_new0(vox_t, number_of_voices);
+	if (voxGetVoices(voices, &number_of_voices) || !number_of_voices)
+		goto exit0;
+
+	DBG(DBG_MODNAME "number_of_voices=%u", number_of_voices);
+
+	/* build speechd_voice */
+	speechd_voice = g_new0(SPDVoice*, number_of_voices + 1);	
+	for (i = 0; i < number_of_voices; i++) {
+		speechd_voice[i] = g_malloc0(sizeof(SPDVoice));
+		if (!vox_to_spd_voice(voices+i, speechd_voice[i]))
+			goto exit0;			
+	}
+	speechd_voice[number_of_voices] = NULL;
+
+	for (i = 0; speechd_voice[i]; i++) {
+		DBG(DBG_MODNAME "speechd_voice[%d]:name=%s, language=%s, variant=%s",
+		    i,
+		    speechd_voice[i]->name ? speechd_voice[i]->name : "null",
+		    speechd_voice[i]->language ? speechd_voice[i]->language : "null",
+		    speechd_voice[i]->variant ? speechd_voice[i]->variant : "null");
+	}
+	
+	DBG(DBG_MODNAME "LEAVE %s", __func__);
+	return TRUE;
+
+ exit0:
+	if (voices) {
+		g_free(voices);
+		voices = NULL;
+	}
+	free_voice_list();
+	return FALSE;
+}
+
+static void load_user_dictionary()
+{
+	DBG(DBG_MODNAME "ENTER %s", __func__);
+	GString *dirname = NULL;
+	GString *filename = NULL;
+	int i = 0;
+	int dictionary_is_present = 0;
+	static guint old_index = G_MAXUINT;
+	guint new_index;
+	const char *language = NULL;
+	const char *region = NULL;
+	ECIDictHand eciDict = eciGetDict(eciHandle);
+
+	new_index = g_atomic_int_get(&locale_index_atomic);
+	if (new_index >= number_of_voices) {
+		DBG(DBG_MODNAME "%s, unexpected index (0x%x)", __FUNCTION__, new_index);
+		return;
+	}
+
+	if (old_index == new_index) {
+		DBG(DBG_MODNAME "LEAVE %s, no change", __FUNCTION__);
+		return;
+	}
+
+	language = voices[new_index].lang;
+	region = voices[new_index].variant;
+
+	/* Fix locale name for French Canadian */
+	if (!strcmp(language, "ca") && !strcmp(region, "FR")) {
```

Comment:
Also, ewwww...
AIUI it's just a bug in voxin that you fixed in 451ba54dff0c84f731aede9e045ed2a0b0831d9f